Src/WindowsServer/WindowsServer.Shared/Implementation/AzureHeartbeatProperties.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    internal class AzureHeartbeatProperties
+    {
+        internal static string HeartbeatPropertyPrefix = "azInst_"; // to ensure no collisions with base heartbeat properties
+
+        /// <summary>
+        /// Expected fields extracted from Azure IMS to add to the heartbeat properties. 
+        /// Set as internal for testing.
+        /// </summary>
+        internal readonly List<string> ExpectedAzureImsFields = new List<string>()
+        {
+            "osType",
+            "location",
+            "name",
+            "offer",
+            "platformFaultDomain",
+            "platformUpdateDomain",
+            "publisher",
+            "sku",
+            "version",
+            "vmId",
+            "vmSize",
+            "subscriptionId",
+            "resourceGroupName"
+        };
+
+        /// <summary>
+        /// Flags that will tell us whether or not Azure VM metadata has been attempted to be gathered or not, and
+        /// if we should even attempt to look for it in the first place. Static to ensure we only ever attempt this
+        /// once for any given consuming application.
+        /// </summary>
+        private bool isAzureMetadataCheckCompleted = false;
```

Comment:
Comment mentions this is static but actually not? Please update comment/code.

Reply:
Done